### ISSUE TITLE
GDAX Heartbeat Channel

### ISSUE THREAD

**Issue by 15step** [coinbase/gdax-node#113]
Hi there.  I'm looking at the documentation and trying to subscribe to the heartbeat channel, with this code:
```

var websocket = new Gdax.WebsocketClient(
    ['BTC-USD'],
    'wss://ws-feed.gdax.com',
    {
        key: API_KEY,
        secret: API_SECRET,
        passphrase: API_PASSPHRASE,
    },
    { heartbeat: true }
)
```
```

  webSocket.on('message', data => {
    console.log(data);
  });

```
But I keep receiving the following error, despite passing BTC-USD
//error
`{ type: 'error',message: 'Failed to subscribe',reason: 'You need to specify at least one product ID for channel heartbeat'
`

I'm guessing this is more to do with my ignorance on how the code works than anything else.  Is there a more appropriate, centralized place to put these comments?

**Comment by nengine** [comment]
I did like below and don't have any issues.  Are you passing API KEY's properly?

```js
const websocket = new Gdax.WebsocketClient(['LTC-USD']);

websocket.on('message', function(data) { 
	/* work with data */ 
	if (data['type'] == 'received') {
		console.log(data);	
	}
});
```

**Comment by fb55** [comment]
This is a shortcoming of the backwards compatibility for heartbeats. If you authenticate the heartbeat will be processed before the other request, resulting in a failed subscription. This library's heartbeat implementation should be switched to use the heartbeat channel instead.

**Comment by nengine** [comment]
Occasionally, GDAX web socket connection will be disconnected even subscribing to heartbeat channel.   @fb55 could you please let me know how to reconnect when it is disconnected?

**Comment by rmm5t** [comment]
Please review #150 for a fix of the heartbeat subscription.

**Comment by rmm5t** [comment]
> could you please let me know how to reconnect when it is disconnected?

@nengine You can reconnect by calling `WebsocketClient#connect()`. For example, in its simplest form:

```js
const client = new Gdax.WebsocketClient(...);
client.on('close', () => client.connect());
```

Of course, you might want to delay the re-connect and/or continue to retry if the connection continues to fail.

**Comment by fb55** [comment]
Fixed in #150
